enigma-swing/src/main/java/cuchaz/enigma/gui/panels/IdentifierPanel.java

```diff
@@ -79,6 +80,36 @@ private void onModifierChanged(AccessModifier modifier) {
 	public void refreshReference() {
 		this.deobfEntry = entry == null ? null : gui.getController().project.getMapper().deobfuscate(this.entry);
 
+		// Prevent IdentifierPanel from being rebuilt if you didn't click off.
+		if (this.lastEntry == this.entry && this.nameField != null) {
+			if (!this.nameField.hasChanges()) {
+
+				final String name;
+
+				// Find what to set the name to.
+				if (this.deobfEntry instanceof MethodEntry && ((MethodEntry) this.deobfEntry).isConstructor()) {
+					// Get the parent of the method if it is a constructor.
+					final ClassEntry parent = ((MethodEntry) this.deobfEntry).getParent();
+
+					if (parent == null) {
+						throw new IllegalStateException("Constructor has no parent?");
```

Comment:
```suggestion
						throw new IllegalStateException("constructor method entry to render has no parent!");
```
just a bit more verbosity here